packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -611,6 +629,24 @@ function build(
     syscall.exit(true, m.serialize(harden(reason)));
   }
 
+  function disavow(presence) {
+    if (!valToSlot.has(presence)) {
+      assert.fail(X`attempt to disavow unknown ${presence}`);
+    }
+    const slot = valToSlot.get(presence);
+    const { type, allocatedByVat } = parseVatSlot(slot);
+    assert.equal(type, 'object', X`attempt to disavow non-object ${presence}`);
+    // disavow() is only for imports: we'll use a different API to revoke
+    // exports, one which accepts an Error object
+    assert.equal(allocatedByVat, false, X`attempt to disavow an export`);
+    valToSlot.delete(presence);
+    slotToVal.delete(slot);
+    const err = harden(Error(`this Presence has been disavowed`));
```

Comment:
While in general I like generality, I still worth asking if you have a use case in mind for giving each disavowal its own error.  There's no logic here to have it vary depending on circumstances.  `disavowedPresences` could be a `WeakSet` that's just checked for membership.  If at some point in the future there's a need to vary the error, you'd have to change to code anyway.

Reply:
Yeah, good point, I first wrote this thinking we could re-use the mechanism for revocations (in which distinct errors are important), and then I thought maybe the caller of `disavow` could/should provide a reason for their decision (or more likely a summary of the thing that was just disavowed) so that later accidental callers could be given a hint as to which taboo object they mistakenly referenced. But then @erights convinced me that referencing a disavowed object should just kill the entire vat, to make it easier for us to remove the feature in the future (by discouraging people from using it: disavowing a shared object in your vat allows any other referencer to kill your vat, so maybe you'll think twice).

I'll change it to use a `WeakSet` and a single error instance as you suggest.